src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -0,0 +1,57 @@
+﻿// <copyright file="ActivityExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// Extension methods on Activity.
+    /// </summary>
+    public static class ActivityExtensions
+    {
+        /// <summary>
+        /// Sets the status of activity execution.
+        /// Activity class in .NET does support 'Status'.
+        /// This extension provides a workaround to store Status as special tags with key name of ot.status_code and ot.status_descriptionkeys.
+        /// Read more about SetStatus here https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-status.
+        /// </summary>
+        /// <param name="activity">Activity instance.</param>
+        /// <param name="status">Activity execution status.</param>
+        /// <param name="description">Status Description.</param>
+        public static void SetStatus(this Activity activity, Status status, string description = null)
```

Comment:
Not sure why we need to pass description in addition to the status?
The following existing approach could be used, right?
https://github.com/open-telemetry/opentelemetry-dotnet/blob/master/test/OpenTelemetry.Tests/Implementation/Trace/StatusTest.cs#L33

Reply:
@cijothomas I asked for that. Basically to save an allocation of a `Status` object just to write a couple of string constants on an `Activity`.

Reply:
Since Status already has a way to add description, taking another parameter is not very appealing for me.

`var status = Status.Unknown.WithDescription("This is an error.");` - This is a good solution right? (Status is a struct)

Reply:
> (Status is a struct)

Doh, I missed that! Sorry @rajkumar-rangaraj @cijothomas is correct it should just be `public static void SetStatus(this Activity activity, Status status)` and if anyone wants to change the description they can use the `WIthDescription` method.

Reply:
@CodeBlanch @cijothomas Thanks for your inputs. I missed looking at `WithDescription` earlier. Now it is fixed.